src/regdesk/states/reg_machine_manager.js

```diff
@@ -0,0 +1,106 @@
+import { LoadingState } from './loading_state.js';
+import { NewSearchState } from './new_search_state.js';
+import { SingleResultState } from './single_result_state.js';
+
+/**
+ * State mchine manager for the reg lookup interface.
+ */
+export class RegMachineManager {
+  /**
+   * Stores the state machine.
+   */
+  #machine;
+
+  /**
+   * Gets the state machine for this machine
+   */
+  get machine() {
+    return this.#machine;
+  }
+
+  /**
+   * Stores the current state of the machine.
+   */
+  #currentState;
+
+  /**
+   * Initializes a new instance of the RegMachineManager class.
+   * @param {RegMachineArgs} regMachineArgs - The standard record for each state class.
+   */
+  constructor(regMachineArgs) {
+    const machine = {
+      initial: NewSearchState.name,
+      states: {},
+    };
+
+    // TODO: Is there a better way to store this?
+    const eventMap = [
```

Comment:
Eh, ultimately not how I would have done things, but I'm not picky. She'll do for our needs for now I suspect.